fixd.js

```diff
@@ -1,14 +1,61 @@
-const fs = require('fs')
-const path = require('path')
-const fixd = require('./src/index')
+'use strict'
+const clone = require('lodash.clonedeep')
 
-// Auto-load built-in plugins
-try {
-  fs.readdirSync(path.resolve(__dirname, './src/plugins')).forEach((f) => {
-    fixd.use(f.replace('.js', ''))
-  })
-} catch (e) {
-  throw new Error('Error loading built-in plugins', e.message)
+const fixd = {
+  /**
+   * Deeply freezes an object
+   * @param {*} obj The object to freeze
+   * @returns {*}
+   */
+  freeze: (obj) => {
+    Object.freeze(obj)
+    Object.getOwnPropertyNames(obj).forEach((prop) => {
```

Comment:
This method throws when called with `null` or `undefined` due to `getOwnPropertyNames`.
```js
fixd.foo = { bar: 'bar' }

fixd.create('foo', (foo) => {
  return null
})
// Object.getOwnPropertyNames(obj).forEach((prop) => {
//        ^
// TypeError: Cannot convert undefined or null to object

fixd.create('foo', (foo) => {
  return undefined
})
// Object.getOwnPropertyNames(obj).forEach((prop) => {
//        ^
// TypeError: Cannot convert undefined or null to object
```